### PR TITLE
process_console: Added documentation

### LINKED ISSUE #1: Document the process console
Tock ships the `process_console` capsule (driver) that provides a shell which allows the management of the currently running processes. The shell commands be quickly displayed by using the `help` command.

Providing a detailed documentation (in the `docs` folder) of the commands and their syntax would help new users get started.

### PR DESCRIPTION
**Pull request by JADarius**
## Pull Request Overview

This pull request adds documentation regarding `process_console` (for issue #1). I also found a typo in the commentary of 
`process_console.rs`.

### TODO or Help Wanted

This pull request still needs a review and suggestions for possible improvement.

### Documentation Updated

- [x] Added the relevant file "Process_Console.md" in `/docs`.

### CODE REVIEW

File: doc/Process_Console.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+ With that console open, you can issue commands. For example, to see all of
+ the processes on the board, use `list`:
+
+ ```text
+ $ tockloader listen
+ Using "/dev/cu.usbserial-c098e513000c - Hail IoT Module - TockOS"
+
+ Listening for serial output.
+ ProcessConsole::start
+ Starting process console
```
Comment (alexandruradovici) [inline doc/Process_Console.md]: I don't think these messages are shown.

**Review by alexandruradovici — CHANGES_REQUESTED** [review]
Did you actually test these commands on any board / qemu or is it the output generated just from looking the source code?

**Comment by alexandruradovici** [comment]
fixes #1

**Comment by JADarius** [comment]
I don't have any of the boards for hands-on testing so the output was all generated just from looking in the source code. For this commit I used qemu but I couldn't make a connection with tockloader to test the commands related to processes, but I managed to check the other commands by forcing them in the source code and changed the output in the documentation.

If you could recommend me any setups to try or how to connect tockloader to qemu that would be of great help.

**Comment by alexandruradovici** [comment]
Use this [project template](https://github.com/Wyliodrin/rust-tock-project) to run Tock using [renode](https://renode.io/). The console should work for the stm32f412 board.

**Comment by JADarius** [comment]
I have used a microbit board to test all the commands and I have completed the documentation with the commands involving processes.

File: doc/Process_Console.md [hunk 2]
```diff
@@ -0,0 +1,588 @@
+ Process Console
+ ===============
+ 
+ `process_console` is a capsule that implements a text console over the UART that allows
```
Comment (alexandruradovici) [inline doc/Process_Console.md]:
```suggestion
 `process_console` is a capsule that implements a small shell over the UART that allows
```

File: doc/Process_Console.md [hunk 3]
```diff
@@ -0,0 +1,588 @@
...
+ a terminal to inspect and control userspace processes.
```
Comment (alexandruradovici) [inline doc/Process_Console.md]:
```suggestion
 a terminal to inspect the kernel and control userspace processes.
```

File: doc/Process_Console.md [hunk 4]
```diff
@@ -0,0 +1,588 @@
...
+
+ Protocol
```
Comment (alexandruradovici) [inline doc/Process_Console.md]:
```suggestion
Commands
```

File: doc/Process_Console.md [hunk 5]
```diff
@@ -0,0 +1,588 @@
...
+ This module provides a simple text-based console to inspect and control
+ which processes are running. The console has eleven commands:
+  - `help` - prints the available commands and arguments
+  - `status` - prints the current system status
+  - `list` - lists the current processes with their IDs and running state
+  - `stop n` - stops the process with name n
+  - `start n` - starts the stopped process with name n
+  - `fault n` - forces the process with name n into a fault state
+  - `terminate n` - terminates the running process with name n, moving to the Terminated state
+  - `boot n` - tries to boot a Terminated process with name n
```
Comment (alexandruradovici) [inline doc/Process_Console.md]: What does `boot` mean? Restart?

File: doc/Process_Console.md [hunk 6]
```diff
@@ -0,0 +1,588 @@
...
+                  Capability));
+ hil::uart::UART::set_client(&usart::USART0, pconsole);
+
+ pconsole.start();
+ ```
+
+ Using ProcessConsole
+ --------------------
+
+ With this capsule properly added to a board's `main.rs` and that kernel
```
Comment (alexandruradovici) [inline doc/Process_Console.md]: What does `that kernel` mean here?

File: doc/Process_Console.md [hunk 7]
```diff
@@ -0,0 +1,588 @@
...
+Which option? [0] 1
+[INFO   ] Using "/dev/ttyACM0 - "BBC micro:bit CMSIS-DAP" - mbed Serial Port".
+[INFO   ] Listening for serial output.
+
+tock$ 
+ ```
+
+ For the examples below we will have 2 processes on the board: blink (which will `blink` all the LEDs that are 
+ connected to the kernel), and `c_hello` (which prints 'Hello World' when the console is started). Also, a micro:bit v2 board was used as support for the commands, so the results may vary on other devices.
+
```
Comment (alexandruradovici) [inline doc/Process_Console.md:93]:
```suggestion
 We will assume that the user has a serial connection to the board, either by using tockloader or another serial port software.
```

**Review by alexandruradovici — CHANGES_REQUESTED** [review]

File: doc/Process_Console.md [hunk 8]
```diff
@@ -0,0 +1,588 @@
...
+                  &mut console::WRITE_BUF,
+                  &mut console::READ_BUF,
+                  &mut console::COMMAND_BUF,
+                  kernel,
+                  Capability));
+ hil::uart::UART::set_client(&usart::USART0, pconsole);
+
+ pconsole.start();
+ ```
+
```
Comment (alexandruradovici) [inline doc/Process_Console.md:47]:
```suggestion
> Note: Using the process console might require allocating more stack to the kernel. This is done by modifying ...
```

File: doc/Process_Console.md (hunk 5 above)
Comment (JADarius) [reply doc/Process_Console.md]: It means restart, I've changed the description.

File: doc/Process_Console.md (hunk 6 above)
Comment (JADarius) [reply doc/Process_Console.md]: It means the Tock kernel, I've changed the documentation.

**Review by alexandruradovici — DISMISSED** [review]

**Comment by alexandruradovici** [comment]
I think you can send this pull request to the upstream repository.

**Comment by JADarius** [comment]
I have submitted a PR: https://github.com/tock/tock/pull/3338

**Comment by alexandruradovici** [comment]
Closed due to https://github.com/tock/tock/pull/3338.

**Comment by alexandruradovici** [comment]
Merged to upstream.

**Comment by alexandruradovici** [comment]
@JADarius Please select a timeslot for the interview https://doodle.com/meeting/participate/id/eZ8gOOEb.

**Pull request closed without merging**